pkg/manifest/subscription.go

```diff
@@ -138,6 +155,17 @@ func subscriptionService(subscriptionOptions subscription.ImageOptions, serviceO
 			commands = append(commands, "/usr/bin/insights-client --register")
 			// insights-client creates the .gnupg directory during boot process, and is labeled incorrectly
 			commands = append(commands, "restorecon -R /root/.gnupg")
+			// register to template if template is specified
+			if subscriptionOptions.TemplateUUID != "" {
+				patchURL := subscriptionOptions.PatchURL
+				if patchURL[len(patchURL)-1] == '/' {
+					patchURL = patchURL[:len(patchURL)-1]
+				}
+				addTemplateURL := fmt.Sprintf("%s/templates/%s/subscribed-systems", patchURL, subscriptionOptions.TemplateUUID)
+				curlToAssociateSystem := fmt.Sprintf("curl -v --retry --proxy %s %s --cert /etc/pki/consumer/cert.pem  --key /etc/pki/consumer/key.pem -X PATCH", subscriptionOptions.Proxy, addTemplateURL)
+				commands = append(commands, curlToAssociateSystem)
+				commands = append(commands, "/usr/sbin/subscription-manager refresh")
```

Comment:
This code block is identical to the one used with `rhc`. Please de-duplicate it into a function and call it from both places instead.

Reply:
Good point